admin_guide/out_of_resource_handling.adoc

```diff
@@ -0,0 +1,312 @@
+independent of its associated grace period, the node reports a condition
+indicating that the node is under pressure.
+
+The following node conditions are defined that correspond to the specified
+eviction signal.
+
+.Node Conditions Related to Low Resources
+[cols="2a,2a,8a",options="header"]
+|===
+
+|Node Condition |Eviction Signal |Description
+
+|`*MemoryPressure*`
+|`*memory.available*`
+|Available memory on the node has satisfied an eviction threshold.
+|===
+
+The node continues to report node status updates at the frequency specified by
+the `node-status-update-frequency` argument, which defaults to *10s*.
+
+[[out-of-resource-oscillation-of-node-conditions]]
+=== Oscillation of Node Conditions
+
+If a node is oscillating above and below a soft eviction threshold, but not
+exceeding its associated grace period, the corresponding node condition
+constantly oscillates between true and false, which can cause poor scheduling
+decisions.
+
+To protect against this oscillation, the following flag is defined to control
+how long the node must wait before transitioning out of a pressure condition.
+
+* `eviction-pressure-transition-period` is the duration for which the node has
+to wait before transitioning out of an eviction pressure condition.
+
+The node ensures that it has not observed an eviction threshold being met for
+the specified pressure condition for the period specified before toggling the
+condition back to *false*.
+
+[[out-of-resource-eviction-of-pods]]
+=== Eviction of Pods
+
+If an eviction threshold is met and the grace period is passed, the node
+initiates the process of evicting pods until it observes the signal going below
+its defined threshold.
+
+The node ranks pods for eviction:
+
+. By their quality of service, and
+. Among those with the same quality of service by the consumption of the
+starved compute resource relative to the pod's scheduling request.
+
+* `BestEffort` pods that consume the most of the starved resource are failed
+first.
+* `Burstable` pods that consume the greatest amount of the starved resource
+relative to their request for that resource are killed first. If no pod has
+exceeded its request, the strategy targets the largest consumer of the starved
+resource.
+* `Guaranteed` pods that consume the greatest amount of the starved resource
+relative to their request are killed first. If no pod has exceeded its request,
+the strategy targets the largest consumer of the starved resource.
+
+A `Guaranteed` pod is guaranteed to never be evicted because of another pod's
+resource consumption. If a system daemon (for example, node, *docker*,
+*journald*, or others) is consuming more resources than were reserved via
+*system-reserved* or *kube-reserved* allocations, and the node only has
+`Guaranteed` pods remaining, then the node must choose to evict a `Guaranteed`
+pod in order to preserve node stability, and to limit the impact of the
+unexpected consumption to other `Guaranteed` pods.
```

Comment:
If there's a way to break up this long sentence, that'd be nice. Not a super big deal.